### PR TITLE
Fixed remove item when depth=0 and wrong gutter calc treeList widget

### PR DESCRIPTION
**Pull request by hanc2006**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes

1. Currently it is not possible to remove an level 0 item from the treeList widget.

```
var item = myList.treeList('get', 'deleted');
item.treeList.remove();

myList.treeList('get', 'deleted');
-> works as expected

myList.treeList('data')
     .filter(item => item.id == 'deleted')
     .map(item => item.label);

-> it does not work as expected, the deleted item returns anyway
```

2. Currently if you remove a child element from the list, and then insert it again the gutter width is calculated incorrectly

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js [hunk 1]
```diff
@@ -348,6 +348,16 @@
...
                 delete that._items[item.id];
+                if(item.depth === 0) {
+                    for(var key in that._items) {
+                        var child = that._items[key];
+                        if(child.parent && child.parent.id === item.id) {
+                            delete that._items[key].treeList;
+                            delete that._items[key];
+                        }
+                    }
+                    that._data = that._data.filter(data => data.id !== item.id)
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js:359]: We do not use arrow functions in the editor due to requirements to support IE11.

```suggestion
                    that._data = that._data.filter(function(data) { return data.id !== item.id})
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js [hunk 2]
```diff
@@ -709,6 +722,16 @@
                 return this._data;
             }
         },
+        item: function(newItem, select) {
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js:725]: What is this function for? I can't see it being part of this fix and there's no mention of it in the PR.

**Review by knolleary — CHANGES_REQUESTED** [review]

File: packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js (hunk 2 above)
Comment (hanc2006) [reply packages/node_modules/@node-red/editor-client/src/js/ui/common/treeList.js:725]: Sorry, this has nothing to do with it. It is a new feature that allows you to add an item at level 0. It was added later in the commit, and I reflected it in the pull request by mistake. I remove it

**Comment by knolleary** [comment]
Thanks. If you'd like to propose new features/apis, then (as per contribution guidelines) please come discuss them in slack before spending lots of time implementing them.

We just need you to sign the CLA before we can merge this - link below.

**Comment by hanc2006** [comment]
Ok, how can I do that?

**Comment by knolleary** [comment]
Scroll down on the PR page to where it says 'Some checks were not successful' - click the 'Details' link next to `licence/cla Pending`.

**Comment by knolleary** [comment]
If you're able to get the cla signed in the next couple of hours, it'll get included in 1.3.4 - otherwise will have to wait for 1.3.5

**Comment by knolleary** [comment]
For future reference, please make sure you run `grunt test-editor` before submitting a PR. I failed to spot the CI failure before merging.

**Pull request merged**